azure-mgmt-cdn/src/main/java/com/microsoft/azure/management/cdn/CdnEndpoint.java

```diff
@@ -0,0 +1,445 @@
+package com.microsoft.azure.management.cdn;
+
+import com.microsoft.azure.management.apigeneration.Fluent;
+import com.microsoft.azure.management.cdn.implementation.EndpointInner;
+import com.microsoft.azure.management.resources.fluentcore.arm.CountryISOCode;
+import com.microsoft.azure.management.resources.fluentcore.arm.models.ExternalChildResource;
+import com.microsoft.azure.management.resources.fluentcore.model.Settable;
+import com.microsoft.azure.management.resources.fluentcore.model.Wrapper;
+
+import java.util.List;
+
+/**
+ * An immutable client-side representation of an Azure CDN endpoint.
+ */
+@Fluent
+public interface CdnEndpoint extends
+        ExternalChildResource<CdnEndpoint, CdnProfile>,
+        Wrapper<EndpointInner> {
+
+    /**
+     * Get the originHostHeader value.
+     *
+     * @return the originHostHeader value
```

Comment:
this is not a user-facing statement, as it is meaningless. i know this is the wording in the auto-gen'd one. We need to make it sound user-facing. i.e "origin host header" not "originHostHeader" value. pls scrub the javadocs with this in mind.